src/lib/operators/materialize.cpp

```diff
@@ -0,0 +1,92 @@
+#include "materialize.hpp"
+
+#include <mutex>
+
+#include "resolve_type.hpp"
+#include "storage/base_column.hpp"
+#include "storage/chunk.hpp"
+#include "storage/create_iterable_from_column.hpp"
+#include "storage/table.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/job_task.hpp"
+
+namespace opossum {
+
+Materialize::Materialize(const std::shared_ptr<const AbstractOperator> in)
+    : AbstractReadOnlyOperator{OperatorType::Materialize, in} {}
+
+const std::string Materialize::name() const { return "Materialize"; }
+const std::string Materialize::description(DescriptionMode description_mode) const { return name(); }
+
+std::shared_ptr<AbstractOperator> Materialize::_on_recreate(
+    const std::vector<AllParameterVariant>& args, const std::shared_ptr<AbstractOperator>& recreated_input_left,
+    const std::shared_ptr<AbstractOperator>& recreated_input_right) const {
+  return std::make_shared<Materialize>(recreated_input_left);
+}
+
+std::shared_ptr<const Table> Materialize::_on_execute() {
+  const auto in_table = _input_left->get_output();
+
+  auto out_table = std::make_shared<Table>(in_table->column_definitions(), TableType::Data);
+
+  std::mutex output_mutex{};
+
+  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
+  jobs.reserve(in_table->chunk_count());
+
+  for (ChunkID chunk_id{0u}; chunk_id < in_table->chunk_count(); ++chunk_id) {
+    auto job_task = std::make_shared<JobTask>([=, &output_mutex]() {
+      const auto in_chunk = in_table->get_chunk(chunk_id);
+
+      // The new chunk is allocated on the same node
+      const auto& alloc = in_chunk->get_allocator();
```

Comment:
I'm aware the std does the same, but I don't like the abbreviation

Reply:
Mhm, it’s used all over our code base :/